entry_user_credentials.go

```diff
@@ -6,325 +6,229 @@ import (
-			CloseCommentIsRequired                   bool `json:"closeCommentIsRequired"`
-			OpenCommentPromptOnBrowserExtensionLink  bool `json:"openCommentPromptOnBrowserExtensionLink"`
-			CloseCommentPrompt                       bool `json:"closeCommentPrompt"`
-			OpenCommentIsRequired                    bool `json:"openCommentIsRequired"`
-			WarnIfAlreadyOpened                      bool `json:"warnIfAlreadyOpened"`
-		} `json:"events"`
-		Data              string                  `json:"data"`
-		Expiration        string                  `json:"expiration"`
-		CheckOutMode      int                     `json:"checkOutMode"`
-		Group             string                  `json:"group"`
-		ConnectionType    ServerConnectionType    `json:"connectionType"`
-		ConnectionSubType ServerConnectionSubType `json:"connectionSubType"`
-		Keywords          string                  `json:"keywords"`
-	}{}
-
-	raw.Id = e.ID
-	raw.Keywords = sliceToKeywords(e.Tags)
-	raw.Description = e.Description
-	raw.RepositoryId = e.VaultId
-	raw.Group = e.EntryFolderPath
-	raw.ConnectionSubType = e.ConnectionSubType
-	raw.ConnectionType = e.ConnectionType
-	raw.Name = e.EntryName
-	sensitiveJson, err := json.Marshal(e.Credentials)
-	if err != nil {
-		return nil, fmt.Errorf("failed to marshal sensitive data. error: %w", err)
-	}
-
-	raw.Data = string(sensitiveJson)
-
-	entryJson, err := json.Marshal(raw)
-	if err != nil {
-		return nil, err
-	}
+const (
+	entryConnectionsEndpoint string = "/api/connections"
+	entryEndpoint            string = "/api/connections/partial"
+	entryPublicEndpoint      string = "/api/v1/vault/{vaultId}/entry/{id}"
+	EntryTypeCredential      string = "Credential"
+	EntrySubTypeDefault      string = "Default"
+)
 
-	return entryJson, nil
+type Entries struct {
+	UserCredential *EntryUserCredentialService
+	Certificate    *EntryCertificateService
+	Host           *EntryHostService
+	Website        *EntryWebsiteService
 }
 
-// UnmarshalJSON implements the json.Unmarshaler interface.
-func (e *EntryUserCredential) UnmarshalJSON(d []byte) error {
-	raw := struct {
-		Data struct {
-			ID                string
-			Description       string
-			Name              string
-			Group             string
-			Username          string
-			ModifiedDate      *ServerTime
-			Keywords          string
-			RepositoryId      string
-			ConnectionType    ServerConnectionType
-			ConnectionSubType ServerConnectionSubType
-		}
-	}{}
-	err := json.Unmarshal(d, &raw)
-	if err != nil {
-		return err
-	}
-
-	e.ID = raw.Data.ID
-	e.EntryName = raw.Data.Name
-	e.ConnectionType = raw.Data.ConnectionType
-	e.ConnectionSubType = raw.Data.ConnectionSubType
-	e.ModifiedDate = raw.Data.ModifiedDate
-	e.Credentials.Username = raw.Data.Username
-	e.Description = raw.Data.Description
-	e.EntryFolderPath = raw.Data.Group
-	e.VaultId = raw.Data.RepositoryId
-
-	e.Tags = keywordsToSlice(raw.Data.Keywords)
-
-	return nil
+type Entry struct {
```

Comment:
Peut etre laisser le nom à EntryUserCredential vu que ca va etre seulement pour les user credentials, mettre un commentaire aussi pour la godoc
```suggestion
// EntryUserCredential represents a DVLS entry/connection.
type EntryUserCredential struct {
```